internal/handlers/pg/msg_getparameter.go

```diff
@@ -101,3 +136,59 @@ func selectParam(document, resDB *types.Document) (doc *types.Document, err erro
 	}
 	return doc, nil
 }
+
+// extractParam is getting parameters showDetails & allParameters from the request.
+func extractParam(document *types.Document) (showDetails, allParameters bool, err error) {
+	getPrm, err := document.Get("getParameter")
+	if err != nil {
+		return false, false, lazyerrors.Error(err)
+	}
+
+	if param, ok := getPrm.(*types.Document); ok {
+		var errType string
+		show, _ := param.Get("showDetails")
+		showDetails, errType = convertBool(show)
+		if errType != "" {
+			s := fmt.Sprintf(`BSON field 'getParameter.showDetails' is the wrong type '%s', `+
+				`expected types '[bool, long, int, decimal, double']`, errType,
+			)
+			return false, false, common.NewErrorMsg(common.ErrTypeMismatch, s)
+		}
+
+		all, _ := param.Get("allParameters")
+		allParameters, errType = convertBool(all)
+		if errType != "" {
+			s := fmt.Sprintf(`BSON field 'getParameter.allParameters' is the wrong type '%s', `+
+				`expected types '[bool, long, int, decimal, double']`, errType,
+			)
+			return false, false, common.NewErrorMsg(common.ErrTypeMismatch, s)
+		}
+	} else {
+		if getPrm == "*" {
+			allParameters = true
+		}
+	}
+	return showDetails, allParameters, nil
+}
+
+// convertBool converts numeric types to bool type.
+func convertBool(v any) (val bool, errType string) {
```

Comment:
We already have this kind of function — `common.GetBoolOptionalParam`

Reply:
there is a double name `getParameter.allParameters` in the error
function only `allParameters`

Reply:
That's ok, just use AssertEqualAltError to check it in tests.

Reply:
The result error text would be kinda `BSON field 'allParameters' is the wrong type 'string', expected types '[bool, long, int, decimal, double]'`